### DISCUSSION TITLE
What do you think about discussions?

### DISCUSSION THREAD

**Discussion by BraisGabin** [detekt/detekt#3971] [General]
After more than 6 months using Github discussions in the project, which is your opinion about it? Should we keep using it? Should we add more categories? Should we remove some?

**Comment by BraisGabin** [comment]
I was eager to use it because I didn't like to have support issues in our issue tracker. But after these months I'm not sure if I like discussions that much. Reasons:

- I miss a "close" button.
- Now if you look for some conversation you need to look at 3 different points: Issues, PRs and discussions.
- Nearly all the discussions have only one thread. So we don't get nearly any benefit from using discussions over issues.

I must say that I like the idea to have our own "stackoverflow" in the repository but I think that it's not working right now.

I'm not talking about closing the discussions but I feel that right now they are not providing us too much value so we should do something.

**Reply by cortinico** [reply]
I'm quite satisfied with it 👍 I think in the long run they will help create a sort of knowledge base that is easier to crawl for users.

> I miss a "close" button.

The Q&A discussions actually have a "answered" button that you can use to mark the discussion as closed.

> Now if you look for some conversation you need to look at 3 different points: Issues, PRs and discussions.

IMHO we should be more proactive and move issue to discussions. Issues should be used only for bug/problems. I feel we receive a lot of support requests in the form of issues that should probably be moved to discussions.

> Nearly all the discussions have only one thread

True, but I also feel that the multi thread nature of discussions makes easier for new-joiners to drop their opinion on discussions. While on a Issue, where the flow of comments is linear, I believe people might feel "intimidated" to drop their opinion.

So overall I believe discussions are great to build up the community. I will try to keep them and eventually revisit the categories or have some sticky discussions on top.

**Reply by schalkms** [reply]
I quite like the new discussion feature. Everything is in one place on GitHub. As Nico mentioned, the search feature is really nice to use. Sure, the discussion feature can be improved. Marking discussions as closed is currently not supported. I don't think that it is huge painpoint, because discussions can at least be _locked_.
Additionally, we have the detekt channel on the official Kotlin Slack.
